File: Java/com.faculdadevincit.designpatterns.commandpattern/src/LuzDesligarCommand.java
public class LuzDesligarCommand implements ICommand {
    private Luz luz;

    public LuzDesligarCommand(Luz luz) {
        this.luz = luz;
    }

    @Override
    public void executar() {
        luz.desligar();
    }

    @Override
    public void desfazer() {
        luz.ligar();
    }

}

File: CSharp/FaculdadeVincit.DesignPatterns.CommandPattern/LuzDesligarCommand.cs
namespace FaculdadeVincit.DesignPatterns.CommandPattern;

public class LuzDesligarCommand : ICommand
{
    public Luz Luz { get; private set; }

    public LuzDesligarCommand(Luz luz)
    {
        Luz = luz;
    }

    public void Executar() => Luz.Desligar();

    public void Desfazer() => Luz.Ligar();
}
